compendium/EM/include/cppmicroservices/em/Event.hpp

```diff
@@ -0,0 +1,163 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+
+#ifndef CppMicroServices_EM_Event_hpp
+#define CppMicroServices_EM_Event_hpp
+
+#include "cppmicroservices/em/EMExport.h"
+
+#include "cppmicroservices/AnyMap.h"
+#include "cppmicroservices/LDAPFilter.h"
+
+namespace cppmicroservices {
+namespace service {
+namespace em {
+
+using EventProperties = std::unordered_map<std::string, Any>;
+
+/**
+ * \defgroup gr_emevent Event
+ * \brief Groups cppmicroservices::service::em::Event related symbols. 
+ */
+
+/**
+ * \ingroup gr_emevent 
+ *
+ * An implementation of the OSGi EventAdmin Event specification.
+ *
+ * The Event is the event object that is both published and subscribed to. Any published events
+ * are of type Event. The event can contain various properties that allow consumers to filter out
+ * specific events and perform certain operations when specific events are received.
+ * 
+ * Event properties are read-only and only allow for consumers to observe them. This is to prevent
+ * event properties being changed by arbitrary pieces of code, ensuring the originality of the
+ * published event.
+ * 
+ * Interactions with events occurs through the EMEvent object directly. See
+ * \c cppmicroservices::em::Constants for information regarding what properties an event can have.
+ */
+class US_em_EXPORT Event
+{
+public:
+  Event() = delete;
+  Event(Event&&) = delete;
+  Event(const Event&&) = delete;
+
+  Event(const Event&) = default;
+
+  virtual ~Event() = default;
+
+  /**
+   * @brief Construct a new Event object with the provided topic and event properties.
+   * 
+   * Topics take the following form:
+   *   topic-description := '*' | topic ( '/\*' )?
+   *   topic := token ( '/' token )*
+   *   token := [A-Za-z0-9_.]
```

Comment:
This comment does not match the implementation in Event.cpp:

  std::string regexExpr = "([A-Za-z0-9_.]+)(\\/[A-Za-z0-9_.]+)*";

"token" needs to have a "+" sign at the end (i.e. token matches 1 or more characters in the set).

Reply:
fixed